spec/spec_helper.cr

```diff
@@ -22,7 +34,7 @@ PERMISSION_FILE = Path[__DIR__, "permissions.json"]
 Spec.before_each do
   DppmRestApi.permissions_config = Fixtures.permissions_config
   DppmRestApi.permissions_config.write_to PERMISSION_FILE
-  DppmRestApi.run Socket::IPAddress::LOOPBACK, DPPM::Prefix.default_dppm_config.port, __DIR__
+  run_server unless already_running
```

Comment:
Why? Also, a method isn't needed.

Reply:
> Why?

Because you only need to start the server once. Starting it multiple times may have unintended side effects. I thought it best to only start it once.

> a method isn't needed.

No, but it makes it neater and more readable, hence more maintainable in the future.

Reply:
Ok. For the method, I don't think so, because we have to look back at another abstraction to see what is the actual code ran.
A method is useful if it's used multiple times.

But, why all of this instead of just having this call before `Spec.before_each`, as done first by me?

Reply:
so option A we can do

```crystal
already_running = false
Spec.before_each do
  DppmRestApi.permissions_config = Fixtures.permissions_config
  DppmRestApi.permissions_config.write_to PERMISSION_FILE
  DppmRestApi.run Socket::IPAddress::LOOPBACK, DPPM::Prefix.default_dppm_config.port, __DIR__ unless already_running
  already_running = true
end
```
or, alternatively --

```crystal
DppmRestApi.permissions_config = Fixtures.permissions_config
DppmRestApi.permissions_config.write_to PERMISSION_FILE
DppmRestApi.run Socket::IPAddress::LOOPBACK, DPPM::Prefix.default_dppm_config.port, __DIR__
Spec.before_each do
  DppmRestApi.permissions_config = Fixtures.permissions_config
  DppmRestApi.permissions_config.write_to PERMISSION_FILE
end
```

Reply:
We could use the latter option, but with less repetition like

```crystal
def reset_config
  DppmRestApi.permissions_config = Fixtures.permissions_config
  DppmRestApi.permissions_config.write_to PERMISSION_FILE
end

reset_config
DppmRestApi.run Socket::IPAddress::LOOPBACK, DPPM::Prefix.default_dppm_config.port, __DIR__

Spec.before_each { reset_config }

Reply:
The latter option sounds good, one less conditional and boolean 👍